name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/actions/InsertAction.java

```diff
@@ -87,6 +93,17 @@ public Status process(BuySellEntry entry, Account account, Portfolio portfolio)
         // security via the dialog)
         process(entry.getPortfolioTransaction().getSecurity());
 
+        // when importing transactions that have already been generated by a savings plan
+        // find the existing item and delete it - it will be created again below
+        if (savingsPlanItem)
+        {
+            DetectDuplicatesAction action = new DetectDuplicatesAction();
+            AccountTransaction existingTransaction = (AccountTransaction) action
+                            .findSavingsPlanTransaction(entry.getAccountTransaction(), account.getTransactions());
+
+            account.deleteTransaction(existingTransaction, client);
```

Comment:
I propose not only to delete the transaction, but add the newly created transaction to the investment plan as well.

Reply:
The investment plan doesn't offer a method to create a transaction from outside. Maybe it is best to find and adjust the existing transaction by copying over differing information like shares, date, note, etc.

Reply:
Added a `addTransaction()` method to `InvestmentPlan` and solved it this way.